cmd/internal/kube/daemonset/daemonset.go

```diff
@@ -75,8 +79,40 @@ func (d *Daemonset) StartProcessInExistingPods() error {
 	return fmt.Errorf("not implemented")
 }
 
-func (d *Daemonset) KubernetesEventsWorker() {
-	// Not implemented
+func (d *Daemonset) KubernetesEventsWorker(done chan struct{}) {
+	for {
+		select {
+		case <-done:
+			return
+		case event := <-d.KubeClient.EventWatch.ResultChan():
+			switch event.Type {
+			case watch.Added:
+				if e, ok := event.Object.(*coreV1.Event); ok {
+					pods, err := d.KubeClient.GetPods([]string{e.InvolvedObject.Name})
+					if err != nil {
+						printer.Errorf("failed to get pod for k8s added event, pod name: %s, error: %v", e.InvolvedObject.Name, err)
+					}
+					if len(pods) == 0 {
+						printer.Errorf("no pods found for k8s added event, pod name: %s", e.InvolvedObject.Name)
+					}
+
+					apidumpArgs, err := d.inspectPodForEnvVars(pods[0])
+					if err != nil {
+						printer.Errorf("failed to inspect pod for env vars, pod name: %s, error: %v", e.InvolvedObject.Name, err)
+					}
+
+					err = d.StartApiDumpProcess(apidumpArgs)
```

Comment:
I don't see where the check for "this doesn't have the right environment variables" goes.  That should probably be a debug log message too.

If there is one but not the other, maybe that deserves a log message?

Reply:
The check for "is Insights already installed as a sidecar" is also missing.